logbook-api/src/main/java/org/zalando/logbook/PathFilter.java

```diff
@@ -0,0 +1,21 @@
+package org.zalando.logbook;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+import org.apiguardian.api.API;
+
+@API(status = EXPERIMENTAL)
+@FunctionalInterface
+public interface PathFilter {
+
+    String filter(final String path);
```

Comment:
The path could be split into its segments, is that better?

Reply:
Could we offer two versions of `filter`, one with `String path` the other with a `List<String> segments`?

Reply:
Sure, and the default implementation of the one with string is to just split into segments and call the other method?

Reply:
Is there some ready-to-use library that we already have available? That default implementation should go in core, not api.

Reply:
@whiskeysierra Or perhaps just having a single filter method for `List<String> segments` is better?

Reply:
Also fine

Reply:
`List<String> segments` turned out to be a dead end, especially since it would not be a noop for the passthrough filter.